packages/devtools/src/App.tsx

```diff
@@ -1,164 +1,116 @@
-                <span
-                  className={`animate-ping absolute inline-flex h-full w-full rounded-full opacity-75 ${connected ? 'bg-green-400' : 'bg-red-400'}`}
-                />
-                <span
-                  className={`relative inline-flex rounded-full h-3 w-3 ${connected ? 'bg-green-500' : 'bg-red-500'}`}
-                />
-              </span>
-            </div>
-          </div>
-
-          <div className="flex flex-1 justify-end">
-            <NavLink
-              to="/"
-              className={({ isActive }) => (isActive ? 'App__route active' : 'App__route')}
-              children={({ isActive }) => {
-                let Icon: JSX.Element = <ChatRegular className="size-5 my-auto mr-1" />;
 
-                if (isActive) {
-                  Icon = <ChatFilled className="size-5 my-auto mr-1" />;
-                }
-
-                return (
-                  <div className="flex">
-                    {Icon}
-                    Chat
-                  </div>
-                );
-              }}
-            />
-
-            <NavLink
-              to="/cards"
-              className={({ isActive }) => (isActive ? 'App__route active' : 'App__route')}
-              children={({ isActive }) => {
-                let Icon: JSX.Element = <CardUiRegular className="size-5 my-auto mr-1" />;
-
-                if (isActive) {
-                  Icon = <CardUiFilled className="size-5 my-auto mr-1" />;
-                }
-
-                return (
-                  <div className="flex">
-                    {Icon}
-                    Cards
-                  </div>
-                );
-              }}
-            />
-
-            <NavLink
-              to="/activities"
-              className={({ isActive }) => (isActive ? 'App__route active' : 'App__route')}
-              children={({ isActive }) => {
-                let Icon: JSX.Element = <SearchRegular className="size-5 my-auto mr-1" />;
-
-                if (isActive) {
-                  Icon = <SearchFilled className="size-5 my-auto mr-1" />;
-                }
-
-                return (
-                  <div className="flex">
-                    {Icon}
-                    Activities
-                  </div>
-                );
-              }}
-            />
-
-            <NavLink
-              to="/logs"
-              className={({ isActive }) => (isActive ? 'App__route active' : 'App__route')}
-              children={({ isActive }) => {
-                let Icon: JSX.Element = (
-                  <DocumentBulletListRegular className="size-5 my-auto mr-1" />
-                );
-
-                if (isActive) {
-                  Icon = <DocumentBulletListFilled className="size-5 my-auto mr-1" />;
-                }
+    return () => {
+      socket.off('activity');
+      socket.disconnect(() => {
+        console.info('Disconnected from server...');
+        setConnected(false);
```

Comment:
I could be wrong but there is a good chance this can lead to setting state after this component unmounted.

Reply:
isn't dismount on the return statement?

Reply:
Oh I see what you mean. Let me clean this up.

Reply:
Logged for fast-follow https://github.com/microsoft/spark.js/issues/18